### PR TITLE
Handle crud view

### PR DESCRIPTION
**Pull request by DarkVisenya**

### CODE REVIEW

File: src/modules/RA/Scene/components/SceneForm.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          <Br />
+          <DisabledInput
+            source="slug"
+            label="view.form.slug"
+          />
+        </>
+      ) : (
+        <>
+          <TextInput source="name" label="view.form.name" />
+          <Br />
```
Comment (mabhub) [inline src/modules/RA/Scene/components/SceneForm.js]: What is the need of this `<Br />`?

File: src/utils/react-admin/helper.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export const convertToSlug = value => {
```
Comment (mabhub) [inline src/utils/react-admin/helper.js]:
```suggestion
export const toSlug = value => {
```

File: src/modules/RA/DataLayer/services/datalayer.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ let viewList = null;
 
 export const fetchViewList = async () => {
   if (viewList === null) {
-    const apiViews =  await Api.request('geolayer/view/');
+    const apiViews =  await Api.request('geolayer/scene/');
 
     // Object transformation from backend format
     // to frontend one.
```
Comment (jrmi) [inline src/modules/RA/DataLayer/services/datalayer.js]: Next lines should not be necessary anymore.

File: src/modules/RA/Scene/components/SceneForm.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                  if (!value) return;
+                  const slugGenerated = convertToSlug(value.target.value);
+                  dispatch(change(REDUX_FORM_NAME, 'slug', slugGenerated));
+                }}
+                {...rest}
+              />
+            )}
+          </FormDataConsumer>
+          <Br />
+          <DisabledInput
```
Comment (jrmi) [inline src/modules/RA/Scene/components/SceneForm.js]: Should be generated but editable here.

File: src/modules/RA/Scene/views/List.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      order: 'ASC',
+    }}
+    exporter={false}
+    bulkActionButtons={<CommonBulkActionButtons />}
+    {...props}
+  >
+    <Datagrid rowClick="edit">
+      <TextField source="name" label="view.form.name" />
+      <TextField source="category" label="view.form.category" />
+      <TextField source="custom_icon" label="view.form.icon" />
```
Comment (jrmi) [inline src/modules/RA/Scene/views/List.js]: Should be an image field ?

File: src/modules/RA/ra-modules.js [hunk 6]
```diff
@@ -24,6 +25,7 @@ export const RES_USERGROUP = 'usergroup';
 // Visu
 export const RES_DATASOURCE = 'datasource';
 export const RES_DATALAYER = 'datalayer';
+export const RES_VIEW = 'view';
```
[the remark this answers is not in the document]
Comment (jrmi) [reply src/modules/RA/ra-modules.js]: I recommend to keep view for now but need a documented decision about global naming.

File: src/services/react-admin/toMultipart.js [hunk 7]
```diff
@@ -1,21 +1,28 @@
...
   /**
    * Manage file upload by converting query content to FormData()
    */
-  if ([CREATE, UPDATE].includes(type) && [RES_DATASOURCE, RES_PICTURE].includes(resource)) {
+  if ([CREATE, UPDATE].includes(type) && modules.includes(resource)) {
     const body = new FormData();
 
     Object.keys(params.data).forEach(key => {
-      if (key === 'file') { return; }
+      if (key === 'file' || key === 'custom_icon') { return; }
```
Comment (jrmi) [inline src/services/react-admin/toMultipart.js]:
```suggestion
      if (filenameProperties.includes(key)) { return; }
```

File: src/modules/RA/Scene/components/SceneForm.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      {isObjectEmpty(record) ? (
+        <>
+          <FormDataConsumer>
+            {({ dispatch, ...rest }) => (
+              <TextInput
+                source="name"
+                label="view.form.name"
+                onChange={value => {
+                  if (!value) return;
+                  const slugGenerated = convertToSlug(value.target.value);
```
[the remark this answers is not in the document]
Comment (DarkVisenya) [reply src/modules/RA/Scene/components/SceneForm.js]: Indeed, this name is better

File: src/utils/react-admin/helper.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+export const toSlug = value => value
+  .toLocaleLowerCase()
+  .normalize('NFD')
```
Comment (mabhub) [inline src/utils/react-admin/helper.js]: image

Isn't this an issue? (ping @jrmi)

Comment (mabhub) [inline src/utils/react-admin/helper.js]: image

Isn't this an issue? (ping @jrmi)

Comment (jrmi) [reply src/utils/react-admin/helper.js]: 88% it's not enough yet, should be modified unless you made a graceful degradation ? i.e. a simple workaround for non compatible navigator.

**Pull request merged**